### ISSUE TITLE
Issue with Slate (1.0.0-beta.12) + Webpack

### ISSUE THREAD

**Issue by artshostak** [Elkfox/Ajaxinate#30]
1. Installed by running:
`yarn add ajaxinate`
`yarn build`

2. Imported in theme.js file by adding the line:
`import Ajaxinate from 'ajaxinate/dist/ajaxinate';`

3. Added the following declaration in the code (with a callback that executes lazy loading of images using bLazy):
``
document.addEventListener("DOMContentLoaded", function() {
  var endlessScroll = new Ajaxinate({
    container: '#AjaxinateLoop',
    pagination: '#AjaxinatePagination',
    method: 'click',
    callback: function() {
      bLazy.revalidate();
    },
  });
});
``
![screenshot 2018-11-19 15 41 38](https://user-images.githubusercontent.com/6820356/48741873-9e588280-ec11-11e8-91f5-9a91b7248156.jpg)

4. Getting the following error:
`_ajaxinate2.default is not a constructor
    at HTMLDocument.eval`

Screenshot:
![screenshot 2018-11-19 15 39 27](https://user-images.githubusercontent.com/6820356/48741792-520d4280-ec11-11e8-9d13-3b46fe4d6c3c.jpg)

Any help at all would be greatly appreciated.

Thanks,

Art

**Comment by Cam** [comment]
@artshostak I think this is a Slate issue, but will need to dig in to test. We're about to finish up a new version of Concrete for Slate 1, and I'll do some testing in that environment once that's up and running.

If you find anything in the meantime, please do add further comments. And thanks for submitting this with useful detail 👍

**Comment by marioloncarek** [comment]
@Cam i managed to init this plugin with solution that @artshostak provided, i was having the same issue, this does not work out of the box with webpack. Im also having the same problem when i want to use method: 'click' . But on method scroll it works normally. Do you maybe have some solutions for that, im getting this errors:


```
Cannot read property 'addEventListener' of null
    at ajaxinateConstructor.addEventListenerForClicking [as addClickListener] (ajaxinate.js?2873:79)
    at ajaxinateConstructor.initializeTheCorrectFunctionsBasedOnTheMethod [as initialize] (ajaxinate.js?2873:62)
    at ajaxinateConstructor.success (ajaxinate.js?2873:119)
```

**Comment by Banjerr** [comment]
I'm currently experiencing this same issue, but I can't seem to get Ajaxinate to work even with the instructions @artshostak provided. I get `Ajaxinate is not defined`. I'm importing it in my `theme.js` file, and initializing it from anywhere does not seem to work.

**Comment by artshostak** [comment]
> Here is how I got it working for the time being —
> 
> 1. Install library
>    `yarn add ajaxinate`
> 2. Import the library
>    `import Ajaxinate from 'ajaxinate/dist/ajaxinate';`
> 3. Add this to your slate.config file to modify the Webpack configuration:
>    `    module: { rules: [ { test: path.resolve('./node_modules/ajaxinate'), use: 'exports-loader?Ajaxinate' } ] },`
> 
> This will require "exports-loader" so add it as well:
> `npm install exports-loader --save-dev`
> 
> 1. Now you can call on the global variable "Ajaxinate" anywhere in your theme code. **The only problem I still have is that when you get to the last page, the following error comes up when triggering the last page of items:**
>    `Uncaught TypeError: Cannot read property 'innerHTML' of undefined at ajaxinateConstructor.success (ajaxinate.js?f0b1:115)`
> 
> Anyone know how to fix that?

The solution to this is to ensure that your Loop and Pagination are siblings as before my Pagination was actually inside the Loop. This DOM structure should do it:
![screenshot 2019-02-19 17 49 18](https://user-images.githubusercontent.com/6820356/53060271-b210c080-346e-11e9-820e-28318be38d00.jpg)

**Comment by eballeste** [comment]
I always get an empty object

**Comment by Cam** [comment]
@eballeste You might be calling Ajaxinate before the object exists in the page. What happens if you console log the container object in place of calling Ajaxinate?

**Comment by eballeste** [comment]
I get a reference to the dom object. 

What I meant was I followed your steps using `exports-loader` to try to import Ajaxinate and what happened before and now, after trying your method, is that I always get an empty object when I console log the Ajaxinate global, for some reason `exports-loader` never worked for me. 

I gave up after a while and added the script into a vendor folder and edited it so that it exported the Ajaxinate object.
